onthisday/__init__.py

```diff
@@ -0,0 +1,216 @@
+    "4": "april",
+    "5": "may",
+    "6": "june",
+    "7": "july",
+    "8": "august",
+    "9": "september",
+    "10": "october",
+    "11": "november",
+    "12": "december",
+}
+
+
+def check(ctx: commands.Context):
+    return lambda x: x.author == ctx.author and x.channel == ctx.channel
+
+
+def highlight_numerical_data(string):
+    return re.sub(r"(([\d{1}],?)+)", r"**\1**", string)
+
+
+def retrieve_above_0(year):
+    return year.strip().isdigit()
+
+
+def columns(years):
+    m = f"{chr(12644)}\n"  # padding
+    f = lambda s: inline(s.zfill(4))
+    for x in range(6, len(years), 6):
+        m += "\n" + (" " * 5).join(map(f, years[x - 6 : x]))
+    return m
+
+
+def get_years(years):
+    return set(years + tuple(map(lambda s: s.zfill(4), years)))
+
+
+def date_suffix(number) -> str:
+    number = int(number)
+    suffixes = {0: "th", 1: "st", 2: "nd", 3: "rd"}
+    for i in range(4, 10):
+        suffixes[i] = "th"
+    return str(number) + suffixes[int(str(number)[-1])]
+
+
+def now() -> datetime.datetime:
+    return datetime.datetime.now()
+
+
+class DateConverter(Converter):
+    """Date converter which uses dateparser.parse()."""
+
+    async def convert(self, ctx: Context, arg: str) -> datetime.datetime:
+        parsed = dateparser.parse(arg)
+        if parsed is None:
+            raise BadArgument("Unrecognized date/time.")
+        if parsed.strftime("%Y") != str(now().strftime("%Y")):
+            raise BadArgument("Please do not specify a year.")
+        return parsed
+
+
+class OnThisDay(commands.Cog):
+    """Find out what happened today, in multiple different years in history."""
+
+    __version__ = "1.0.0"
+    __author__ = "Kreusada"
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+        if 719988449867989142 in self.bot.owner_ids:
+            with contextlib.suppress(RuntimeError, ValueError):
+                self.bot.add_dev_env_value(self.__class__.__name__.lower(), lambda x: self)
+
+    def format_help_for_context(self, ctx: commands.Context) -> str:
+        context = super().format_help_for_context(ctx)
+        return f"{context}\n\nAuthor: {self.__author__}\nVersion: {self.__version__}"
+
+    def cog_unload(self):
+        self.bot.loop.create_task(self.session.close())
+        if 719988449867989142 in self.bot.owner_ids:
+            with contextlib.suppress(KeyError):
+                self.bot.remove_dev_env_value(self.__class__.__name__.lower())
+
+    async def red_delete_data_for_user(self, **kwargs):
+        """Nothing to delete"""
+        return
+
+    async def run_otd(
+        self, ctx: commands.Context, month_number, date_number, year, _random: bool = False
+    ):
+        """Oh my, is this scruffy code... but it WORKS!"""
+        try:
+            async with self.session.get(
+                ENDPOINT.format(f"{month_number}/{date_number}")
+            ) as session:
+                content = await session.json()
```

Comment:
Would probably be worth checking `session.response` to make sure that the status code is 200
```suggestion
                if session.status != 200:
                    return await ctx.maybe_send_embed("The api could not be reached. Sorry")
                content = await session.json()
```